Pico8Emulator/unit/cart/CartridgeUnit.cs

```diff
@@ -0,0 +1,453 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Pico8Emulator.lua;
+using Pico8Emulator.unit.graphics;
+using GraphicsUnit = Pico8Emulator.unit.graphics.GraphicsUnit;
+
+namespace Pico8Emulator.unit.cart {
+	public class CartridgeUnit : Unit {
+		public Cartridge Loaded;
+		private DateTime startTime;
+
+		public CartridgeUnit(Emulator emulator) : base(emulator) {
+			
+		}
+
+		public override void Update() {
+			base.Update();
+			Loaded?.Interpreter.CallIfDefined("_update");
+		}
+
+		public void Draw() {
+			if (Loaded == null) {
+				return;
```

Comment:
Would be nice to log a warning somewhere so the user knows they haven't loaded any carts yet. This way it is silently "failing" and can be confusing for people to identify what's wrong